torch/distributed/_tensor/placement_types.py

```diff
@@ -150,20 +150,28 @@ def _shard_tensor(
         """
         shard and scatter a tensor on a mesh dimension (use coordinate
         0 on the mesh dimension as source of truth)
+
+        if tensor is on meta, then just allocate an empty tensor on the desired device based on the device mesh's
+        device_type
         """
         my_coordinate = mesh.get_coordinate()
         num_chunks = mesh.size(dim=mesh_dim)
 
         if my_coordinate is None:
             # if rank is not part of mesh, we simply return an empty tensor
-            return tensor.new_empty(0, requires_grad=tensor.requires_grad)
+            return tensor.new_empty(
+                0, device=mesh.device_type, requires_grad=tensor.requires_grad
+            )
 
         scatter_list, pad_sizes = self._split_tensor(
             tensor, num_chunks, with_padding=True, contiguous=True
         )
 
-        output = torch.empty_like(scatter_list[my_coordinate[mesh_dim]])
-        mesh.scatter(output, scatter_list, mesh_dim=mesh_dim)
+        output = torch.empty_like(
+            scatter_list[my_coordinate[mesh_dim]], device=mesh.device_type
```

Comment:
I think we don't need to pass device to this `empty_like`  call, `empty_like` would just reuse the same device with its input

Reply:
I've removed it